internal/report/snapshot/snapshot_buffers.go

```diff
@@ -86,11 +86,15 @@ func (l *LogBuffer) Add(ld plog.Logs) {
 	}
 }
 
+// LogsMarshaller marshals logs to a byte representation
+type LogsMarshaller interface {
+	MarshalLogs(ld plog.Logs) ([]byte, error)
+}
```

Comment:
Any reason not to just make this a func?
```suggestion
type MarshalLogs func(plog.Logs) ([]byte, error)
```

Reply:
We can, it's basically equivalent, but it makes it just slightly more awkward to use:

Current:
```go
payload, err := logBuffer.ConstructPayload(&plog.ProtoMarshaler{}, nil, nil)
```

As function:
```go
payload, err := logBuffer.ConstructPayload((&plog.ProtoMarshaler{}).MarshalLogs, nil, nil)
```

I don't really mind either way, so if you think there's a benefit to the function approach, I'm happy to change it.

Reply:
Ah, I see the motivation then.